src/main/java/macrobase/analysis/outlier/OutlierClassifier.java

```diff
@@ -0,0 +1,19 @@
+package macrobase.analysis.outlier;
+
+import macrobase.analysis.result.OutlierClassificationResult;
+import macrobase.conf.MacroBaseConf;
+import macrobase.datamodel.Datum;
+
+public abstract class OutlierClassifier {
+
+    private OutlierClassificationResult lastResult;
+
+    public OutlierClassifier(MacroBaseConf conf) {}
+    public abstract void observe(Datum datumToClassify);
+
+    public abstract OutlierClassificationResult classify();
```

Comment:
The proposal:
  * `classify()` returns an `OutlierClassificationResult`
  * `OutlierClassificationResult` implements a method `boolean isOutlier()` and a method `Datum getDatum()`

You don't need to keep the entire stream around to implement this.